compiler/rustc_target/src/asm/x86.rs

```diff
@@ -294,13 +295,14 @@ def_regs! {
         zmm29: zmm_reg = ["zmm29", "xmm29", "ymm29"] % x86_64_only,
         zmm30: zmm_reg = ["zmm30", "xmm30", "ymm30"] % x86_64_only,
         zmm31: zmm_reg = ["zmm31", "xmm31", "ymm31"] % x86_64_only,
-        k1: kreg = ["k1"],
-        k2: kreg = ["k2"],
-        k3: kreg = ["k3"],
-        k4: kreg = ["k4"],
-        k5: kreg = ["k5"],
-        k6: kreg = ["k6"],
-        k7: kreg = ["k7"],
+        k0: kreg0 = ["k0"],
+        k1: kreg, kreg0 = ["k1"],
+        k2: kreg, kreg0 = ["k2"],
+        k3: kreg, kreg0 = ["k3"],
+        k4: kreg, kreg0 = ["k4"],
+        k5: kreg, kreg0 = ["k5"],
+        k6: kreg, kreg0 = ["k6"],
+        k7: kreg, kreg0 = ["k7"],
```

Comment:
I find the `kreg0` register group name quite surprising. Intuitively I expected this register class to include just the `k0` register, rather than being `kreg + k0`. In retrospective it makes sense, but I felt there was definitely a surprise factor of sorts. I wonder if there is a name which would have an obvious interpretation.

As a side note, preexisting, and probably half my fault, but isn't the naming of this group inconsistent with the other groups? We can't fix this now, can we?

Reply:
I don't have a better ideas for the register class name.

I don't think the naming is too inconsistent. `kreg` is what you usually want most of the time.

Reply:
> I don't have a better ideas for the register class name.

Hmm… `kreg_full` or `kreg_all`, perhaps?

> I don't think the naming is too inconsistent. kreg is what you usually want most of the time.

Ah, I meant in relation to other classes like `abcd_reg`, `xmm_reg`, `ymm_reg` et al. If the convention of separating `reg` from types of register, `kreg` would be `k_reg`. Nothing we can do here anymore, though.

Reply:
The rule is consistent but idiosyncratic: kreg, vreg, preg, then xmm_reg, x87_reg, etc.
Like
```rust
if class.len() <= 1 {
    format!("{class}reg")
else {
    format!("{class}_reg")
}
```